week2/jalalhm/test2.js

```diff
@@ -0,0 +1,122 @@
+"use strict";
+
+const fs = require('fs');
+
+const TODOS_FILENAME = process.env["TODOS_FILENAME"] || "todos.json";
+
+let myTodoList = []
+
+// Helper function definitions
+
+let readTodosFile = (filename) => {
+    return new Promise((resolve, reject) => {
+        fs.readFile(filename, "utf8", (err, data) => {
+            if (err) {
+                throw err
+                reject()
+            }
+            //console.log(`Successfully read the file: ${filename}`)
+            resolve(data)
+
+        })
+    })
+};
+
+let writeTodosFile = (filename, dataAsString) => {
+    return new Promise((resolve, reject) => {
+        fs.writeFile(TODOS_FILENAME, dataAsString, (err) => {
+            if (err) {
+                errors(err)
+            };
+            //console.log(`Saved todos file: ${filename}`);
+            resolve();
+        })
+    })
+}
+
+let addTodoItem = (filename, todoList, itemText) => {
+    todoList.push({
+        message: itemText
+    })
+    writeTodosFile(filename, JSON.stringify(myTodoList)).then(() => console.log("Done adding todo."));
```

Comment:
You need a `.catch()` in case an error occurs or the promise fails (this is true of all of your promise calls that use a `.then()`).